src/schema.ts

```diff
@@ -0,0 +1,45 @@
+import { gql } from "apollo-server";
+
+export const typeDefs = gql`
+  type Customer {
+    id: String!
+    name: String!
+    email: String!
+  }
+
+  type LendingRecord {
+    customer: Customer!
+    lentDate: String!
+    returnDate: String
```

Comment:
won't the return date be also optional?

Reply:
From my understanding of GraphQL if a field doesn't have the "!" it can be nullable which in my case serves as optional, so if we don't have a return date it is just null, which can be used to check if a book is returned later.

Reply:
ah yeah that's right! I was looking at the .ts extension of the file and was thinking about TS type here. nvm my comment, but thanks for taking the time to explain it 🥳